src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -265,134 +232,172 @@ public void setUrl(String url) {
          * @return the headers
          */
         public Map<String, String> getHeaders() {
-            return headers;
+            return Collections.unmodifiableMap(this.headers);
         }
 
+        /**
+         * @return the list of options which are ignored
+         */
         public List<String> getOptionsInProperties() {
-            return optionsInProperties;
+            return Collections.unmodifiableList(this.optionsInProperties);
         }
 
+        /**
+         * @param option the option
+         */
         public void addOptionsInProperties(String option) {
             this.optionsInProperties.add(option);
         }
 
+        /**
+         * @return the maximum transfer rate
+         */
         public int getLimitRate() {
             return limitRate;
         }
 
+        /**
+         * @param limitRate the maximum transfer rate
+         */
         public void setLimitRate(String limitRate) {
             String unit = limitRate.substring(limitRate.length() - 1, limitRate.length()).toLowerCase();
             int value = Integer.parseInt(limitRate.substring(0, limitRate.length() - 1).toLowerCase());
             switch (unit) {
+            case "k":
+                this.limitRate = value * 128;
+                break;
             case "m":
-                value = value * 1000;
+                this.limitRate = value * 128000;
                 break;
             case "g":
-                value = value * 1000 * 1000;
+                this.limitRate = value * 128000000;
```

Comment:
I think I would go with the formula instead of the end result. The java compiler will probably do the calculation on compile time anyway.